### PR TITLE
Add aarch64 wheel build support

### PR DESCRIPTION
**Pull request by odidev**
Added aarch64 wheel build support.
Related to #3413. @postrational could you please review this PR?

### CODE REVIEW

File: .github/workflows/release_linux_aarch64.yml [hunk 1]
```diff
@@ -0,0 +1,80 @@
+name: LinuxRelease_aarch64
+
+on:
+  schedule:
+    # Run weekly on Monday 00:00
+    - cron:  '00 00 * * MON'
+  push:
+    branches: [master, rel-*]
+  pull_request:
+    branches: [rel-*]
```
Comment (jcwchen) [inline .github/workflows/release_linux_aarch64.yml:10]:
```suggestion
    branches: [rel-*, master]  # TODO remove master before merge
```
Could you please add master trigger for testing? Thank you.

File: .github/workflows/manylinux/entrypoint.sh [hunk 2]
```diff
@@ -34,11 +38,16 @@ cd $ONNX_PATH
...
-PYTHON_COMAND="/opt/python/"${PY_VER}"/bin/python"
+if [ `uname -m` == 'aarch64' ]; then
+ PIP_COMMAND="$PY_VERSION -m pip install --no-cache-dir"
+ PYTHON_COMAND="$PY_VERSION"
+else
+ declare -A python_map=( ["3.6"]="cp36-cp36m" ["3.7"]="cp37-cp37m" ["3.8"]="cp38-cp38" ["3.9"]="cp39-cp39")
+ declare -A python_include=( ["3.6"]="3.6m" ["3.7"]="3.7m" ["3.8"]="3.8" ["3.9"]="3.9")
+ PY_VER=${python_map[$PY_VERSION]}
+ PIP_COMMAND="/opt/python/${PY_VER}/bin/pip install --no-cache-dir"
+ PYTHON_COMAND="/opt/python/"${PY_VER}"/bin/python"
```
Comment (jcwchen) [inline .github/workflows/manylinux/entrypoint.sh:49]: L43, L49
nit: this should be my typo: PYTHON_COMMAND

File: .github/workflows/manylinux/entrypoint.sh [hunk 3]
```diff
@@ -76,4 +85,4 @@ fi
 rm -f dist/*-linux*.whl
 
 echo "Succesfully build wheels:"
-find . -type f -iname "*-manylinux*.whl"
\ No newline at end of file
+find . -type f -iname "*-manylinux*.whl"
```
Comment (jcwchen) [inline .github/workflows/manylinux/entrypoint.sh:88]: I thought the wheel name for aarch64 would be something else other than -manylinux? Also L85 might be not needed for aarch64

File: .github/workflows/release_linux_aarch64.yml (hunk 1 above)
Comment (odidev) [reply .github/workflows/release_linux_aarch64.yml]: Done.

File: .github/workflows/manylinux/entrypoint.sh (hunk 2 above)
Comment (odidev) [reply .github/workflows/manylinux/entrypoint.sh]: Corrected.

File: .github/workflows/manylinux/entrypoint.sh (hunk 3 above)
Comment (odidev) [reply .github/workflows/manylinux/entrypoint.sh:88]: It is required as before auditwheel the aarch64 wheels will be generated with name "onnx-1.8.1-cp39-cp39-linux_aarch64.whl" and  auditwheel repair creates new wheel  file with name "onnx-1.8.1-cp39-cp39-manylinux2014_aarch64.whl".

File: .github/workflows/manylinux/entrypoint.sh (hunk 2 above)
Comment (jcwchen) [reply .github/workflows/manylinux/entrypoint.sh]: Also L66 and L68 for PIP_COMMAND. Thank you!

**Comment by odidev** [comment]
@jcwchen, could you please merge this PR. Thanks in advance.

**Comment by jcwchen** [comment]
Could @onnx/sig-archinfra-approvers please review this PR? Thanks.

**Comment by odidev** [comment]
Could you please review this PR? Thanks

**Comment by askhade** [comment]
@odidev : Please fix pipeline failures:
image

File: .github/workflows/release_linux_aarch64.yml [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+          ${{ env.img }} \
+          bash -exc '\
+          yum install -y sudo && \
+          sudo chmod +x .github/workflows/manylinux/entrypoint.sh && \
+          sudo .github/workflows/manylinux/entrypoint.sh ${{ env.py }} manylinux2014_aarch64 ${{ github.event_name }}'
+    - name: Test wheel with Python ${{ matrix.python-version }}
+      run: |
+        docker run --rm -v ${{ github.workspace }}:/ws:rw --workdir=/ws \
+          ${{ env.img }} \
+          bash -exc '${{ env.py }} -m pip install virtualenv && ${{ env.py }} -m venv .env && \
```
[the remark this answers is not in the document]
Comment (odidev) [reply .github/workflows/release_linux_aarch64.yml]: By using virtual environment we are making already installed dependencies available for different tasks.

File: .github/workflows/release_linux_aarch64.yml [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+
+on:
+  schedule:
+    # Run weekly on Monday 00:00
+    - cron:  '00 00 * * MON'
+  push:
+    branches: [master, rel-*]
+  pull_request:
+    branches: [rel-*, master]  # TODO remove master before merge
+
```
[the remark this answers is not in the document]
Comment (odidev) [reply .github/workflows/release_linux_aarch64.yml:12]: Done.

File: .github/workflows/release_linux_aarch64.yml [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+jobs:
+  build:
+    runs-on: ubuntu-latest
```
[the remark this answers is not in the document]
Comment (odidev) [reply .github/workflows/release_linux_aarch64.yml:16]: Done.

File: .github/workflows/release_linux_aarch64.yml [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+    steps:
+    - uses: actions/checkout@v2
+    - uses: docker/setup-qemu-action@v1
+    - name: Checkout submodules
+      shell: bash
+      run: |
+        auth_header="$(git config --local --get http.https://github.com/.extraheader)"
+        git submodule sync --recursive
+        git -c "http.extraheader=$auth_header" -c protocol.version=2 submodule update --init --force --recursive --depth=1
+    - name: Install python dependencies
```
[the remark this answers is not in the document]
Comment (odidev) [reply .github/workflows/release_linux_aarch64.yml]: Done.

File: .github/workflows/release_linux_aarch64.yml [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+      run: |
+        docker run --rm -v ${{ github.workspace }}:/ws:rw --workdir=/ws \
+          ${{ env.img }} ${{ env.py }} -m pip install -U numpy protobuf==3.11.3
+    - name: Install dependencies
```
[the remark this answers is not in the document]
Comment (odidev) [reply .github/workflows/release_linux_aarch64.yml:41]: Done.

File: .github/workflows/release_linux_aarch64.yml [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+    - uses: actions/upload-artifact@v1
+      with:
+        name: wheels
+        path: dist
+
+    - name: Upload wheel to TestPyPI weekly
+      if: (github.event_name == 'schedule') # Only triggered by weekly event
+      run: |
+        python -m pip install -q twine
+        twine upload --verbose dist/*.whl --repository-url https://test.pypi.org/legacy/ -u ${{ secrets.TESTPYPI_USERNAME }} -p ${{ secrets.TESTPYPI_PASSWORD }}
```
[the remark this answers is not in the document]
Comment (odidev) [reply .github/workflows/release_linux_aarch64.yml:98]: Done.

File: .github/workflows/release_linux_aarch64.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (askhade) [reply .github/workflows/release_linux_aarch64.yml]: git actions does let you do this across tasks... you dont need virtual env for that

File: .github/workflows/release_linux_aarch64.yml [hunk 10]
```diff
@@ -0,0 +1,137 @@
+name: LinuxRelease_aarch64
+
+on:
+  schedule:
+    # Run weekly on Monday 00:00
+    - cron:  '00 00 * * MON'
+  push:
+    branches: [master, rel-*]
+  pull_request:
+    branches: [rel-*, master]  # TODO remove master before merge
```
Comment (jcwchen) [inline .github/workflows/release_linux_aarch64.yml:10]: nit: this #TODO comment can be removed now

File: .github/workflows/release_linux_aarch64.yml [hunk 11]
```diff
@@ -0,0 +1,137 @@
...
+  workflow_dispatch:
+
+jobs:
+  build:
+    if: github.event_name != 'pull_request' || startsWith( github.base_ref, 'rel-') || contains( github.event.pull_request.labels.*.name, 'run release CIs')
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        # the different python versions for building wheels
+        python-version: [cp37-cp37m, cp38-cp38, cp39-cp39]
```
Comment (jcwchen) [inline .github/workflows/release_linux_aarch64.yml:20]: How about Python 3.6?

File: .github/workflows/release_linux_aarch64.yml [hunk 12]
```diff
@@ -0,0 +1,137 @@
...
+      # and installing the dependencies inside that \
+      # so that we can use installed dependencies.
+    - name: Install dependencies
+      run: |
+        docker run --rm -v ${{ github.workspace }}:/ws:rw --workdir=/ws \
+          ${{ env.img }} \
+          bash -exc '${{ env.py }} -m pip install virtualenv && ${{ env.py }} -m venv .env && \
+          source .env/bin/activate && \
+          ${{ env.py }} -m pip install -U numpy protobuf==3.11.3 && \
+          yum install -y protobuf-compiler protobuf-devel
```
Comment (jcwchen) [inline .github/workflows/release_linux_aarch64.yml:48]: I assume the protobuf version from yum package is not 3.11.3, right? I think a better way is to install protobuf from source like this: https://github.com/onnx/onnx/blob/31f0fd6103e231664d1cfd3b25264f672f5de960/.github/workflows/manylinux/entrypoint.sh#L18

**Comment by askhade** [comment]
@odidev : can you please resolve conflicts?

**Comment by askhade** [comment]
@odidev : we can check this PR in once the conflicts are resolved

**Comment by askhade** [comment]
@odidev : FYI - The code freeze date for ONNX 1.10 is July 15th (today) since this PR is reviewed and ready we can wait until July 20th. Otherwise this PR will miss ONNX 1.10 release.

**Comment by odidev** [comment]
> @odidev : can you please resolve conflicts?

Done.

File: .github/workflows/release_linux_aarch64.yml (hunk 12 above)
Comment (odidev) [reply .github/workflows/release_linux_aarch64.yml:48]: Yes, I checked it, it is not required as it is already installed.

File: .github/workflows/release_linux_aarch64.yml (hunk 11 above)
Comment (odidev) [reply .github/workflows/release_linux_aarch64.yml]: Done.

File: .github/workflows/release_linux_aarch64.yml (hunk 10 above)
Comment (odidev) [reply .github/workflows/release_linux_aarch64.yml]: Removed.

**Review by jcwchen — APPROVED** [review]
Thanks for updates! Overall looks good to me. Let me reopen this PR to test this new CI.

**Review by askhade — APPROVED** [review]

**Pull request merged**